packages/client-sdk-web/src/internal/auth-client.ts

```diff
@@ -0,0 +1,81 @@
+import {auth} from '@gomomento/generated-types-webtext';
+import {CredentialProvider, MomentoLogger, GenerateApiToken} from '..';
+import {version} from '../../package.json';
+import {Configuration} from '../config/configuration';
+import {Request, UnaryInterceptor, UnaryResponse} from 'grpc-web';
+import {Header, HeaderInterceptorProvider} from './grpc/headers-interceptor';
+import {_GenerateApiTokenRequest} from '@gomomento/generated-types-webtext/dist/auth_pb';
+import {cacheServiceErrorMapper} from '../errors/cache-service-error-mapper';
+import {_GenerateApiTokenResponse} from '@gomomento/common';
+import Never = _GenerateApiTokenRequest.Never;
+import Expires = _GenerateApiTokenRequest.Expires;
+
+export interface AuthClientProps {
+  configuration: Configuration;
+  credentialProvider: CredentialProvider;
+}
+
+export class InternalWebGrpcAuthClient<
+  REQ extends Request<REQ, RESP>,
+  RESP extends UnaryResponse<REQ, RESP>
+> {
+  private readonly clientAuthWrapper: auth.AuthClient;
+  private readonly interceptors: UnaryInterceptor<REQ, RESP>[];
+  private readonly logger: MomentoLogger;
+
+  /**
+   * @param {AuthClientProps} props
+   */
+  constructor(props: AuthClientProps) {
+    this.logger = props.configuration.getLoggerFactory().getLogger(this);
+    const headers = [new Header('Agent', `nodejs:${version}`)];
+    this.interceptors = [
+      new HeaderInterceptorProvider<REQ, RESP>(
+        headers
+      ).createHeadersInterceptor(),
+    ];
+    this.logger.debug(
+      `Creating control client using endpoint: '${props.credentialProvider.getControlEndpoint()}`
+    );
+    console.log(
+      `\n\n\nCreating control client with endpoint: ${props.credentialProvider.getControlEndpoint()}\n\n\n`
+    );
+
+    this.clientAuthWrapper = new auth.AuthClient(
+      `https://${props.credentialProvider.getControlEndpoint()}`,
+      null,
+      {
+        unaryInterceptors: this.interceptors,
+      }
+    );
+  }
+
+  public async generateApiToken(
+    sessionToken: string,
+    validUntilSeconds?: number
```

Comment:
I don't know of the best way to do this, but it would be nice if this could be something like

```
expiresAt: SomeSeconds | Never
```

But I am not sure the best way to expose this to customers other than having 2 separate methods, which is not ideal

Reply:
Created: https://github.com/momentohq/client-sdk-nodejs/issues/394